mindsdb/integrations/libs/ml_exec_base.py

```diff
@@ -282,6 +283,21 @@ def learn(
 
             training_data_df = result['result']
 
+            databases_meta = self.database_controller.get_dict()
+            data_integration_meta = databases_meta[integration_name]
+            # TODO improve here. Suppose that it is view
+            if data_integration_meta['type'] == 'project':
+                data_integration_ref = {
+                    'type': 'view'
+                }
+            else:
+                data_integration_ref = {
+                    'type': 'integration',
+                    'id': data_integration_meta['id']
+                }
+
+        problem_definition = {'target': target}
```

Comment:
this clears  problem_definition (except of target)

Reply:
fixed